src/test/java/com/google/devtools/build/lib/bazel/repository/CompressedTarFunctionTest.java

```diff
@@ -64,6 +66,26 @@ public void testDecompressWithPrefix() throws Exception {
     archiveDescriptor.assertOutputFiles(outputDir, INNER_FOLDER_NAME);
   }
 
+  /**
+   * Test decompressing a tar.gz file, with some entries being renamed during the
+   * extraction process.
+   */
+  @Test
+  public void testDecompressWithRenamedFiles() throws Exception {
```

Comment:
can we add a test case where strip_prefix is also present, to test that rename_files happens before strip_prefix?

Reply:
Done.